backend/src/test/java/org/cryptomator/hub/license/LicenseHolderTest.java

```diff
@@ -62,14 +68,16 @@ public void testValidDBTokenSet() {
 
 			holder.init();
 
-			Mockito.verify(validator, Mockito.times(1)).validate("token", "42");
+			Mockito.verify(validator, Mockito.times(2)).validate("token", "42");
```

Comment:
Shouldn't `validator.validate(...)` be only called once?

Reply:
The problem here is when I switch to an constructor based initialization, I get an 

```
Error running Quarkus: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:113)
        at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:70)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        ... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
        ... 15 more
Caused by: jakarta.enterprise.inject.CreationException: Error creating synthetic bean [3a974b6a18ac399f675913d732c105426414d370]: jakarta.enterprise.inject.CreationException: Synthetic bean instance for org.hibernate.Session not initialized yet: org_hibernate_Session_e591f1489b359bbca9551b0cf0d3d3d28d72a4a5
        - a synthetic bean initialized during RUNTIME_INIT must not be accessed during STATIC_INIT
        - RUNTIME_INIT build steps that require access to synthetic beans initialized during RUNTIME_INIT should consume the SyntheticBeansRuntimeInitBuildItem
        at org.hibernate.Session_3a974b6a18ac399f675913d732c105426414d370_Synthetic_Bean.doCreate(Unknown Source)
        at org.hibernate.Session_3a974b6a18ac399f675913d732c105426414d370_Synthetic_Bean.create(Unknown Source)
        at org.hibernate.Session_3a974b6a18ac399f675913d732c105426414d370_Synthetic_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:113)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:37)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:34)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:32)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:34)
        at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:21)
        at org.hibernate.Session_3a974b6a18ac399f675913d732c105426414d370_Synthetic_ClientProxy.arc$delegate(Unknown Source)
        at org.hibernate.Session_3a974b6a18ac399f675913d732c105426414d370_Synthetic_ClientProxy.find(Unknown Source)
        at io.quarkus.hibernate.orm.panache.common.runtime.AbstractJpaOperations.findById(AbstractJpaOperations.java:179)
        at org.cryptomator.hub.entities.Settings.findById(Settings.java)
        at org.cryptomator.hub.entities.Settings.get(Settings.java:52)
        at org.cryptomator.hub.license.LicenseHolder.<init>(LicenseHolder.java:40)
        at org.cryptomator.hub.license.LicenseHolder_ClientProxy.<init>(Unknown Source)
        at org.cryptomator.hub.license.LicenseHolder_Bean.proxy(Unknown Source)
        at org.cryptomator.hub.license.LicenseHolder_Bean.get(Unknown Source)
        at org.cryptomator.hub.license.LicenseHolder_Bean.get(Unknown Source)
        ... 40 more
Caused by: jakarta.enterprise.inject.CreationException: Synthetic bean instance for org.hibernate.Session not initialized yet: org_hibernate_Session_e591f1489b359bbca9551b0cf0d3d3d28d72a4a5
        - a synthetic bean initialized during RUNTIME_INIT must not be accessed during STATIC_INIT
        - RUNTIME_INIT build steps that require access to synthetic beans initialized during RUNTIME_INIT should consume the SyntheticBeansRuntimeInitBuildItem
        at org.hibernate.Session_3a974b6a18ac399f675913d732c105426414d370_Synthetic_Bean.createSynthetic(Unknown Source)
        ... 60 more
```

when `Settings.get();` is called, which is probably the reason for the current implementation using `@PostConstruct`. 

We could probably work around this with an entity manager, but I'm not sure if it actually makes the implementation "better" in the end.